samples/xamarin-forms/LoginApp/LoginApp/ViewModels/SignUpViewModel.cs

```diff
@@ -0,0 +1,87 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Linq;
+using System.Reactive;
+using Acr.UserDialogs;
+using LoginApp.ViewModels.Abstractions;
+using ReactiveUI;
+using ReactiveUI.Fody.Helpers;
+using ReactiveUI.Validation.Abstractions;
+using ReactiveUI.Validation.Contexts;
+using ReactiveUI.Validation.Extensions;
+using Splat;
+
+namespace LoginApp.ViewModels
+{
+    public class SignUpViewModel : ViewModelBase, ISupportsValidation
+    {
+        private readonly Func<string, bool> _isDefined = value => !string.IsNullOrEmpty(value);
+        private readonly IUserDialogs _dialogs;
+
+        [Reactive] public string UserName { get; set; }
+
+        [Reactive] public string Password { get; set; }
+
+        [Reactive] public string ConfirmPassword { get; set; }
+
+        public ReactiveCommand<Unit, bool> SignUp { get; }
+
+        public ValidationContext ValidationContext { get; } = new ValidationContext();
+
+        public SignUpViewModel(IScreen hostScreen = null, IUserDialogs dialogs = null)
+            : base(hostScreen)
+        {
+            _dialogs = dialogs ?? Locator.Current.GetService<IUserDialogs>();
+            SignUp = ReactiveCommand.Create(SignUpImpl);
```

Comment:
Another curious thing. It seems that if you pass `ValidationContext.Valid` to `ReactiveCommand.Create`, the bound button will stay disabled until the form becomes valid. So perhaps worth doing something like this:
```cs
SignUp = ReactiveCommand.Create(SignUpImpl, ValidationContext.Valid);
```
And if it works, the `ValidationContext.GetIsValid()` check could be omitted from `SignUpImpl`.

Reply:
It won't work since the IsValid property is not an `IObservable<bool>` but `bool` property. But that's a good point and we could improve it.

Reply:
Yeah the `IsValid` property is `bool`, but there is also `Valid` property which is `IObservable<bool>`, see https://github.com/reactiveui/ReactiveUI.Validation/blob/7e9278e0e4c896fd1eb541140f348c3b62909391/src/ReactiveUI.Validation/Contexts/ValidationContext.cs#L88

Although `IsValid` is declared as OAPH, so `WhenAnyValue(x => x.ValidationContext.IsValid)` should probably do the trick as well.

Reply:
Oh I see, you are right!